### DISCUSSION TITLE
Formatting cells to have three decimal places

### DISCUSSION THREAD

**Discussion by AnnePdM** [JanMarvin/openxlsx2#1171] [Q&A] [answered]
Is there anyone out there who can help me with formatting a column to have three decimal places. I currently have this line of code within my file definition and I would like the number format to be 0.000
`wb_add_cell_style("BMSAT Item Summary", dims = paste0("C2:D",nrow(items)), numFmtId = "2", horizontal = "right") %>%`
I have tried to create a format (`create_numfmt`) but I think I must be doing something wrong because it is not working

**Comment by JanMarvin** [comment, marked as answer]
Why don’t you use `wb_add_numfmt()` you should be able to adjust the example.

**Reply by AnnePdM** [reply]
Can't believe I didn't notice that function! Many thanks, it worked.

**Reply by JanMarvin** [reply]
Perfect!

**Reply by JanMarvin** [reply]
A longer example, I was kinda surprised by the question, but after looking at our documentation I realized that we never mention the use of `wb_add_numfmt(numfmt = "#,0")`. So below you find two ways how to assign style `"0.000"` to a cell containing `pi`. I had to look up the long way, because I don't use it anymore, since the short way is so convenient. But internally the short version uses something similar to the long version.

``` r
library(openxlsx2)

### the short way
wb <- wb_workbook()$add_worksheet("short")$
  add_data(x = pi)$
  add_numfmt(numfmt = "0.000")

### the long way
wb$add_worksheet("long")$
  add_data(x = pi)

# create numfmt and add it to workbook
new_numfmt <- create_numfmt(numFmtId = wb$styles_mgr$next_numfmt_id(), formatCode = "0.000")
wb$styles_mgr$add(new_numfmt, "test")

# create cell style with numfmt and add it to workbook
new_xf <- create_cell_style(
  numFmtId = wb$styles_mgr$get_numfmt_id("test")
)
wb$styles_mgr$add(new_xf, "new_xf")

# assign numfmt style to cell
wb$set_cell_style(sheet = current_sheet(), dims = "A1", wb$styles_mgr$get_xf_id("new_xf"))

if (interactive()) wb$open()
```
